From c596ea609944380f7bc66a56b5344b5c44d5163e Mon Sep 17 00:00:00 2001
From: Andre McCurdy <armccurdy@gmail.com>
Date: Thu, 12 Nov 2015 15:40:00 -0800
Subject: [PATCH] BSEAV/linux/driver/usermode/Makefile: add -nostdinc to
 BCM_DRIVER_CFLAGS

Reduce chances of accidentally building kernel space code with
user space header files. Change applies to both ARM and MIPS.

Signed-off-by: Andre McCurdy <armccurdy@gmail.com>
---
 BSEAV/linux/driver/usermode/Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/BSEAV/linux/driver/usermode/Makefile b/BSEAV/linux/driver/usermode/Makefile
index 8d7f958..6dc71e2 100644
--- a/BSEAV/linux/driver/usermode/Makefile
+++ b/BSEAV/linux/driver/usermode/Makefile
@@ -113,6 +113,10 @@ vpath %.c ${BCM_OBJ_ROOT}/interrupt_table
 
 BCM_DRIVER := $(BCM_OBJ_PATH)/bcmdriver.ko
 BCM_DRIVER_CFLAGS += -DKBUILD_MODNAME=bcmdriver -DLINUX -D__KERNEL__ -DMODULE -mlong-calls
+
+NOSTDINC_FLAGS := -nostdinc -isystem $(shell $(CC) -print-file-name=include)
+BCM_DRIVER_CFLAGS += $(NOSTDINC_FLAGS)
+
 ifeq ($(LINUX_VER_GE_3),y)
 BCM_DRIVER_CFLAGS += -include linux/kconfig.h
 else
-- 
1.9.1

